src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -451,3 +452,59 @@ export const handleFormAuthRedirect = [
   }),
   _handleFormAuthRedirect,
 ] as RequestHandler[]
+
+/**
+ * Handler for validating the eServiceId of a given form
+ *
+ * @returns 200 with eserviceId validation result
+ * @returns 400 when there is an error on the authType of the form
+ * @returns 400 when the eServiceId of the form does not exist
+ * @returns 404 when form with given ID does not exist
+ * @returns 500 when the title of the fetched login page does not exist
+ * @returns 500 when database error occurs
+ * @returns 500 when the url for the login page of the form could not be generated
+ * @returns 502 when the login page for singpass could not be fetched
+ */
+export const handleValidateFormEsrvcId: RequestHandler<
+  { formId: string },
+  PublicFormAuthValidateEsrvcIdDto | ErrorDto
+> = (req, res) => {
+  const { formId } = req.params
+  return FormService.retrieveFormById(formId)
+    .andThen((form) => {
+      // NOTE: Because the check is based on parsing the html of the returned webpage,
+      // And because MyInfo login is beyond our control, we coerce MyInfo to SP.
+      // This is valid because a valid MyInfo eserviceId is also a valid SP eserviceId
+      switch (form.authType) {
+        case AuthType.MyInfo:
+          return validateMyInfoForm(form).andThen((form) =>
+            SpcpFactory.createRedirectUrl(AuthType.SP, formId, form.esrvcId),
+          )
+        case AuthType.SP:
+        case AuthType.CP:
+          return validateSpcpForm(form).andThen((form) =>
+            SpcpFactory.createRedirectUrl(form.authType, formId, form.esrvcId),
+          )
```

Comment:
I don't think we should be supporting Corppass here, as the error page is different from the Singpass error page
![Screenshot 2021-04-28 at 10 35 48 AM](https://user-images.githubusercontent.com/29480346/116338191-f39b1400-a80d-11eb-92b4-fa67c1e2db1f.png)

Reply:
Good point! I can take a (brief) look at this and the html structure of the page just to check if it's possible to validate using html and string matching. 

If it's not, I'll probably just lump this together with the default case!

Reply:
err don't waste time on that imo, we have no intention of supporting corppass e-service ID validation anytime soon. we would much rather push for a proper validation endpoint.